tests/test_nml_generation.py

```diff
@@ -23,5 +23,25 @@ def test_generate_nml():
     assert test_result_nml == expected_nml
 
 
+def test_no_default_values_written():
+    # read and write the test file
+    with open("testdata/nml_without_default_values.nml", "r") as file:
+        test_nml = parse_nml(file)
+        with open("testoutput/nml_without_default_values.nml", "wb") as output_file:
+            write_nml(file=output_file, nml=test_nml)
+
+
+    # read the written testfile and compare the content
+    with open("testdata/nml_without_default_values.nml", "r") as file:
+        test_nml = parse_nml(file)
+        with open("testoutput/nml_without_default_values.nml", "r") as output_file:
+            test_result_nml = parse_nml(output_file)
+
+            assert test_nml == test_result_nml, "The testdata file and the testoutput file do not have the same content."
+
+    # test if both files have the same length
+    assert os.path.getsize("testdata/nml_without_default_values.nml") == os.path.getsize("testoutput/nml_without_default_values.nml"), "The testdata file and the testoutput file do not have the same length."
```

Comment:
No because the parsed NML's will defnetly include the default values. But they should not be written to the disk. Thus the test_file has all default values deleted. So if both files have the same size it ensures that the written NML does not include the default values either.